aztec_macros/src/transforms/functions.rs

```diff
@@ -749,8 +749,10 @@ fn create_loop_over(var: Expression, loop_body: Vec<Statement>) -> Statement {
     // What will be looped over
 
     // - `hasher.add({ident}[i] as Field)`
-    let for_loop_block =
-        expression(ExpressionKind::Block(BlockExpression { statements: loop_body }));
+    let for_loop_block = expression(ExpressionKind::Block(BlockExpression {
+        is_unsafe: false,
+        statements: loop_body,
+    }));
```

Comment:
I'm not a fan of `is_unsafe` being a property of any block - I think it should be separate like how a `comptime {}` block is a separate construct. It should be a separate expression kind with another expression kind inside instead.

Reply:
Good point! I'm working on refactoring this. I also noticed that with the current rules, this parses:

```rust
if true unsafe {

}
```

because an if's condition expects a block, and the above is just a block marked as unsafe :-)